dali/operators/segmentation/random_mask_pixel.cc

```diff
@@ -0,0 +1,203 @@
+  USE_OPERATOR_MEMBERS();
+};
+
+RandomMaskPixelCPU::RandomMaskPixelCPU(const OpSpec &spec)
+    : Operator<CPUBackend>(spec),
+      seed_(spec.GetArgument<int64_t>("seed")),
+      has_value_(spec.ArgumentDefined("value")) {
+  if (has_value_) {
+    DALI_ENFORCE(!spec.ArgumentDefined("threshold"),
+      "Arguments ``value`` and ``threshold`` can not be provided together");
+  }
+}
+
+bool RandomMaskPixelCPU::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                    const workspace_t<CPUBackend> &ws) {
+  const auto &in_masks = ws.template InputRef<CPUBackend>(0);
+  int nsamples = in_masks.size();
+  auto in_masks_shape = in_masks.shape();
+  int ndim = in_masks_shape.sample_dim();
+  output_desc.resize(1);
+  output_desc[0].shape = uniform_list_shape(nsamples, {ndim});
+  output_desc[0].type = TypeTable::GetTypeInfo(DALI_INT64);
+
+  foreground_.resize(nsamples);
+  value_.clear();
+  threshold_.clear();
+
+  for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+    foreground_[sample_idx] = spec_.template GetArgument<int>("foreground", &ws, sample_idx);
+  }
+
+  if (spec_.ArgumentDefined("value")) {
+    value_.resize(nsamples);
+    for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+      value_[sample_idx] = spec_.template GetArgument<int>("value", &ws, sample_idx);
+    }
+  } else {
+    threshold_.resize(nsamples, 0.0f);
+    if (spec_.ArgumentDefined("threshold")) {
+      for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+        threshold_[sample_idx] = spec_.template GetArgument<float>("threshold", &ws, sample_idx);
+      }
+    }
+  }
+  return true;
+}
+
+template <typename T>
+void RandomMaskPixelCPU::RunImplTyped(workspace_t<CPUBackend> &ws) {
+  const auto &in_masks = ws.template InputRef<CPUBackend>(0);
+  auto &out_pixel_pos = ws.template OutputRef<CPUBackend>(0);
+  int nsamples = in_masks.size();
+  auto in_masks_shape = in_masks.shape();
+  int ndim = in_masks_shape.sample_dim();
+  auto masks_view = view<const T>(in_masks);
+  auto pixel_pos_view = view<int64_t>(out_pixel_pos);
+  auto& thread_pool = ws.GetThreadPool();
+
+  if (rng_.empty()) {
+    for (int i = 0; i < thread_pool.size(); i++) {
+      rng_.emplace_back(seed_ + i);
+    }
+  }
+  assert(rng_.size() == static_cast<size_t>(thread_pool.size()));
+
+  if (rle_.empty()) {
+    rle_.resize(thread_pool.size());
+  }
+  assert(rle_.size() == static_cast<size_t>(thread_pool.size()));
+
+  for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+    thread_pool.AddWork(
+      [&, sample_idx](int thread_id) {
+        auto &rng = rng_[thread_id];
+        auto mask = masks_view[sample_idx];
+        auto pixel_pos = pixel_pos_view[sample_idx];
+        const auto &mask_sh = mask.shape;
+        if (foreground_[sample_idx]) {
+          int64_t flat_idx = -1;
+          auto &rle_mask = rle_[thread_id];
+          if (has_value_) {
+            T value = static_cast<T>(value_[sample_idx]);
```

Comment:
I have mixed feelings about that one - consider that:
input is uint8
value is 256 
magically, you select 0 as foreground.
Actually:
```c++
if (static_cast<int>(value) != values_[sample_idx])
   // behave as if there was no foregound at all
```

Reply:
Done